client/my-sites/checkout/checkout/domain-details-form.jsx

```diff
@@ -195,19 +206,33 @@ export class DomainDetailsForm extends PureComponent {
 	};
 
 	handleChangeEvent = event => {
-		// Resets the state field every time the user selects a different country
+		this.shouldFocusAddress = false;
+
+		// Every time the user selects a different country
 		if ( event.target.name === 'country-code' ) {
+			// Reset the state field
 			this.formStateController.handleFieldChange( {
 				name: 'state',
 				value: '',
 				hideError: true,
 			} );
 
+			// If the phone field is empty, set the prefix to the countryCode
 			if ( ! formState.getFieldValue( this.state.form, 'phone' ) ) {
 				this.setState( {
 					phoneCountryCode: event.target.value,
 				} );
 			}
+
+			// Focus on the first address field
+			// whenever the user initiates a country change.
+			// If RegionAddressFieldsets is mounted we just need to find the stored inputRef
+			const inputRef = this.inputRefs[ 'address-1' ] || null;
+			inputRef && inputRef.focus();
+
+			// but if RegionAddressFieldsets isn't mounted (because there's no valid countryCode)
+			// we tell it to focus when it next mounts
+			this.shouldFocusAddress = true;
```

Comment:
We could make this borderline self-documenting by pulling it out into a function like `this.focusAddressField()`

Reply:
I like it. Will do.